### PR TITLE
Add OpenAI-powered interview question generation with auth, service object, and edge case handling - branch: Setup for open ai

### PR DESCRIPTION
**Pull request by KMPWeiland**
### Type of Change
- [x] feature ⛲
- [ ] bug fix 🐛
- [x] documentation update 📃
- [ ] styling 🎨
- [x] refactor 🧑‍💻
- [x] testing 🧪

### Description

This PR introduces functionality for generating interview questions using the OpenAI API. The logic first checks whether questions already exist for a specific user's job application. If they do, it returns those from the database. If not, it generates new questions via OpenAI, saves them, and returns the newly created records.
Key Updates:
* Implemented fetch_or_create action in InterviewQuestionsController for better semantic clarity
* Added InterviewQuestionGeneratorService to encapsulate interview question logic
* Integrated Pundit for authorization (authorize, skip_authorization)
* Secured controller with before_action :authenticate_user
* Added nested GET route: ⁠/api/v1/users/:user_id/job_applications/:job_application_id/interview_questions/fetch_or_create
* Update the README
* Added request specs covering:
    * Returning existing questions
    * Generating new questions
    * Handling job app not found
    * Handling OpenAI failures and malformed responses

 
Design Considerations:
	•	Implemented a custom ⁠fetch_or_create action instead of overloading standard CRUD actions
	•	Ensureed OpenAI generation happens only once per job application
	•	Ensured proper error handling for all edge cases

### Share the reason(s) for this pull request
Added interview question generation via OpenAI with a fetch_or_create endpoint that returns existing questions or generates new ones when needed.

### What questions do you have/what do you want feedback on?
Thoughts on the custom endpoint: A GET request with custom fetch_or_create action.
 I was torn between a POST/create vs. a. GET/show pattern. The goal is retrieval (GET), but would have the side effect of question generation if questions aren't available (and GET endpoints should avoid side-effects, POST more appropriate).  

### Related Tickets
n/a

### Screenshots (if applicable):
n/a

### Added Test?
- [x] Yes 🫡

- Request/Controller Specs for the InterviewQuestionsController with comprehensive coverage of happy paths and edge cases
- Additional tests for error handling on the OpenAiGateway

- [ ] No 🙅

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [x] All tests (previous and new) pass 🥳
- [ ] This PR includes a Migration and I have notified the PM so they can run the migration after the PR is merged.

### How to QA this change:
1.	Send a GET request to ⁠/api/v1/users/:user_id/job_applications/:job_application_id/interview_questions/fetch_or_create
2.	Verify thfirst request generates and returns 10 interview questions
3.	Verify subsequent requests return the same questions without calling OpenAI again
4.	Test with invalid job application ID to confirm proper error handling

### CODE REVIEW

File: app/services/interview_question_generator_service.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class InterviewQuestionGeneratorService
```
Comment (JonoSommers) [inline app/services/interview_question_generator_service.rb:1]: Really great work finding a way to generate new questions only when no existing questions were found.

**Review by JonoSommers — APPROVED** [review]
Really great work all around, Kristin! I love how you found a way to generate new questions if none exist, that’s an excellent approach. The testing is thorough, and I really like the error handling and the sad path test that covers the potential failure of the OpenAI API.

**Review by kevin-newland — APPROVED** [review]
Awesome job takling this big issue. I also like how you updated the readme with clear documentation for your changes!

File: app/controllers/api/v1/interview_questions_controller.rb [hunk 2]
```diff
@@ -1,34 +1,28 @@
...
 
   private
 
-  def interview_question_params
-    params.require(:interview_question).permit(:question, :job_application_id)
-  rescue ActionController::ParameterMissing
-    nil
+  def set_job_application
+    @job_application = JobApplication.find_by(id: params[:job_application_id])
+    unless @job_application
```
[the remark this answers is not in the document]
Comment (KMPWeiland) [reply app/controllers/api/v1/interview_questions_controller.rb:22]: skip_authorization is a built-in Pundit method. The app uses the Pundit gem for authorization, but I included ⁠skip_authorization here to tell Pundit that I'm intentionally bypassing the permission check when a job application isn't found. This prevents Pundit from raising its own error and allows the custom "Job application not found" error response to be returned instead.

File: app/serializers/interview_question_serializer.rb [hunk 3]
```diff
@@ -6,17 +6,14 @@ def self.format_questions(questions, response_id)
...
-        {index: index,
-        type: "interview_question",
-        attributes: {
-          question: question,
-          # job_application_id: question["job_application_id"],
+        {
+          index: index, 
+          type: "interview_question",
+            attributes: {
+              question: question.respond_to?(:question) ? question.question : question
```
[the remark this answers is not in the document]
Comment (KMPWeiland) [reply app/serializers/interview_question_serializer.rb:13]: This serializer formats both InterviewQuestion model instances (when fetching from database) and raw question strings (when newly generated from OpenAI).  I just used index(1) so they were more readable on the FE -- the question list should start at 1 instead of the default 0.

I'll add these notes to the README.

[the remark this answers is not in the document]
Comment (KMPWeiland) [reply app/serializers/interview_question_serializer.rb:13]: Sorry, I missed your previous question. The OpenAI bit wasn't too bad actually. I mostly struggled with getting up to speed on the existing functionality, especially as regards the authentications using JWT Tokens and the Pundit gem.    I also went back and forth a lot on the architecture -- what kind of request should it be and will the FE be sending info in the params or the in the body.
